TROUBLESHOOTING.md

```diff
@@ -0,0 +1,492 @@
+Spyder Troubleshooting Guide
```

Comment:
Is this file referenced from somewhere else? If not, my vote is to remove it and only leave the one in the wiki.

The problem with having two places with the same content is that (inevitably) they will diverge at some point in time.

Reply:
> Is this file referenced from somewhere else? If not, my vote is to remove it and only leave the one in the wiki.

Fair enough; I figured this could be an objection, which is why I made it the last commit and not reference it elsewhere to make it easy to revert/remove if needed :) 

> The problem with having two places with the same content is that (inevitably) they will diverge at some point in time.

I wasn't going to include it for the same reason, but I ended up reconsidering it in the end after I specifically tailored it to only include the essentially static general troubleshooting content, which shouldn't change substantially at least over a reasonable timescale so it won't ever be that out of date, and it also includes a disclaimer and link at the top to the full/live version of the guide. Conveniently, the sections I included conveniently form the first and last chunk of the full version, and it uses the exact same Github markdown syntax, so all that needs to be done to update it every few months, if need be, is a simple copy and paste (which I am more than happy to do before each release, assuming there are actually any changes to the relevant sections). 

On the other hand, the benefits of having it there aren't enormous, but the main reason is that troubleshooting help is included with the program itself, same as the other documentation, readme, contributing, etc (which in some cases could also be argued to be duplicative, and much harder to maintain due to different syntax and structure), and so if the user doesn't have access to the Internet (or Github, in some countries), they are still able to get help. As a secondary benefit, it also makes it (and the linked guide) more visible/easily accessible to those browsing the source locally or on Github, and serves as a tracked, publicly accessible backup of the key sections in case Github wikis ever go down or go away.

However, I'm certainly far from wed to it, and already mentally prepared myself for the possible breakup :). With the above in mind, let me know what you'd like me to do with it. Thanks!